src/controllers/organizationController.ts

```diff
@@ -215,6 +217,77 @@ export default class OrganizationController {
     });
   }
 
+  public addDelegate(
+    req: Request
+  ): Promise<
+    // tslint:disable-next-line:max-union-size
+    | IResponseErrorValidation
+    | IResponseErrorInternal
+    | IResponseErrorForbiddenNotAuthorized
+    | IResponseErrorNotFound
+    | IResponseErrorConflict
+    | IResponseSuccessRedirectToResource<Organization, Organization>
+  > {
+    return withUserFromRequest(req, async user => {
+      const userPermissions = accessControl.can(user.role);
+      if (!userPermissions.createOwn(Resource.DELEGATE).granted) {
+        return ResponseErrorForbiddenNotAuthorized;
+      }
+      const errorOrMaybeOrganization = await getOrganizationFromUserEmail(
+        user.email
+      );
+      return withResultOrInternalError(
+        errorOrMaybeOrganization,
+        async maybeOrganization => {
+          if (isSome(maybeOrganization)) {
+            return ResponseErrorConflict(
+              "You are already associated to an organization"
+            );
+          }
+          const errorResponseOrSuccessResponse = await addDelegate(
+            req.params.ipaCode,
+            user.email
+          ).run();
```

Comment:
not needed for this pr: we should write a version of `toExpressHandler` that works with TaskEither instead of promises so we can avoid calling `await / run()` (ever!) and just chain / map return values (or errors)